### DISCUSSION TITLE
Quasar Vue 3 - correct use ref

### DISCUSSION THREAD

**Discussion by jandresmedina** [quasarframework/quasar#16244] [General - Components / Directives / etc] [answered]
Hello, I try to make use of the ref in vue 3, but I receive an error of $ref not defined, any idea how to solve it? I'm new to vue 3
perform a : console.log(table.value) and the response was Proxy(Object) {…}
Exactly I get error in this line of code:

  this.$refs.table.value.filteredSortedRows.map(row => columnsAttendance.map(col => wrapCsvValue( ....)
Error message: properties of undefined (reading '$refs')

```
<template>
<q-table ref="table">
</qtable>
</template>

<script>
import { defineComponent, ref, onMounted } from 'vue'

export default defineComponent({
name: 'PageAttendanceList',

setup () {
const table = ref(null)

return { 
table,
exportFiltered () {
console.log(table.value)

const content = [columnsAttendance.map(col => wrapCsvValue(col.label))].concat(
this.$ref.table.filteredSortedRows.map(row => columnsAttendance.map(col => wrapCsvValue(
typeof col.field === 'function'
? col.field(row)
: row[col.field === void 0 ? col.name : col.field],
col.format
)).join(','))
).join('\r\n')
    }
}
</script>
```

**Comment by jandresmedina** [comment]
I made the change but for some reason I get this error message on this line of code:
table.value.filteredSortedRows.map(row => columnsAttendance.map(..)

Error message: Cannot read properties of undefined (reading 'map')

**Comment by metalsadman** [comment, marked as answer]
if options api, that should be `this.$refs` answered you on your other thread https://github.com/quasarframework/quasar/discussions/16234

**Reply by jandresmedina** [reply]
works for me, amazing, I'm sure it will help many people who are going through this inconvenience.

thanks for your time, you're great
